pkgs/applications/graphics/vpuppr/default.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, alsa-lib
+, gcc-unwrapped
+, git
+, libGLU
+, libglvnd
+, libpulseaudio
+, perl
+, zlib
+, udev
+, libX11
+, libXcursor
+, libXext
+, libXfixes
+, libXi
+, libXinerama
+, libXrandr
+, libXrender
+, fetchzip
+, autoPatchelfHook
+, copyDesktopItems
+, makeDesktopItem
+, fetchurl
+}:
+stdenv.mkDerivation rec {
+  pname = "vpuppr";
+  version = "0.9.0";
+
+  src = fetchzip {
+    url = "https://github.com/virtual-puppet-project/vpuppr/releases/download/0.9.0/vpuppr_0.9.0_linux.zip";
+    sha256 = "sha256-JKiVM1N/NOa5StqUG+BV7ZvNZmDfi5jNKaeWJLy/pTA=";
+  };
+
+  icon = fetchurl {
+    url = "https://github.com/virtual-puppet-project/vpuppr/raw/0.9.0/assets/osfgd_icon.png";
+    sha256 = "sha256-2SdiUCfHLhbMKzpfyulp/qDGShNwCvMXotkBH0I7VDM=";
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Virtual Puppet Project";
+      desktopName = "Virtual Puppet Project";
+      comment = "VTuber application made with Godot 3.5";
+      exec = "vpuppr";
+      icon = "vpuppr_icon.png";
+      categories = [ "Graphics" "AudioVideo" "Recorder" ];
+      terminal = false;
+    })
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  libs = [
+    alsa-lib
+    gcc-unwrapped.lib
+    git
+    libGLU
+    libglvnd
+    libpulseaudio
+    perl
+    zlib
+    udev
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  propagatedBuildInputs = libs;
+
+  buildPhase = ''
+    chmod +x vpuppr.x86_64
+  '';
```

Comment:
I suppose you can change the permissions in the `installPhase`.
(I think you can use `install -Dm555`, etc.)